src/api/DatabaseAccess/CourseDbContext.cs

```diff
@@ -247,19 +247,21 @@ public List<FeeCaps> GetFeeCaps()
 
         public List<Provider> SuggestProviders(string query)
         {
-            return Providers.FromSql(@"
+            var providers =  Providers.FromSql(@"
 SELECT * FROM (
     SELECT ""provider"".*, COUNT(*) AS cnt
     FROM ""provider""
     JOIN ""course"" ON ""course"".""ProviderId"" = ""provider"".""Id""  OR ""course"".""AccreditingProviderId"" = ""provider"".""Id""
     WHERE (to_tsvector('english', ""provider"".""Name"") @@ to_tsquery('english', quote_literal(@query) || ':*')) IS TRUE
     OR (to_tsvector('english', ""provider"".""ProviderCode"") @@ to_tsquery('english', quote_literal(@query) || ':*')) IS TRUE
     GROUP BY ""provider"".""Id"") AS sub
-ORDER BY lower(""Name"") <> lower(@query) ASC, ""cnt"" DESC
-LIMIT @limit",
-            new NpgsqlParameter("@query", query.Replace(@"\","")),
-            new NpgsqlParameter("@limit", 5))
+ORDER BY lower(""Name"") <> lower(@query) ASC, ""cnt"" DESC",
+            new NpgsqlParameter("@query", query.Replace(@"\","")))
             .ToList();
+
+            var matchedProviders = providers.Where(p => p.Name.ToLower().StartsWith(query.ToLower()));
+            var nonMatchedProviders = providers.Where(p => !p.Name.ToLower().StartsWith(query.ToLower()));
+            return matchedProviders.Concat(nonMatchedProviders).Take(5).ToList();
```

Comment:
I would document in a code comment why we need this (considering we'll need to read and re-implement in teh railz)

// This is to make sure that people typing exact names where the words are matching many records in the fuzzy search will get the matching provider at the top (because `StartsWith()` matches)
// specific example: "Teach North..." - a prefix of a few providers, also matches on fuzzy search for things like "Teaching Alliance (North)",
// this was resulting in "Teach North..." not making the top-5 limit and being effectively invisible.